Headlines/src/Categories/UIImageEffects.swift

```diff
@@ -0,0 +1,331 @@
+ without specific prior written permission from Apple. Except as
+ expressly stated in this notice, no other rights or licenses, express or
+ implied, are granted by Apple herein, including but not limited to any
+ patent rights that may be infringed by your derivative works or by other
+ works in which the Apple Software may be incorporated.
+ The Apple Software is provided by Apple on an "AS IS" basis. APPLE MAKES
+ NO WARRANTIES, EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION THE
+ IMPLIED WARRANTIES OF NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR
+ A PARTICULAR PURPOSE, REGARDING THE APPLE SOFTWARE OR ITS USE AND
+ OPERATION ALONE OR IN COMBINATION WITH YOUR PRODUCTS.
+ IN NO EVENT SHALL APPLE BE LIABLE FOR ANY SPECIAL, INDIRECT, INCIDENTAL
+ OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) ARISING IN ANY WAY OUT OF THE USE, REPRODUCTION,
+ MODIFICATION AND/OR DISTRIBUTION OF THE APPLE SOFTWARE, HOWEVER CAUSED
+ AND WHETHER UNDER THEORY OF CONTRACT, TORT (INCLUDING NEGLIGENCE),
+ STRICT LIABILITY OR OTHERWISE, EVEN IF APPLE HAS BEEN ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ EA1002
+ 5/3/2013
+ */
+
+//
+//  UIImage.swift
+//  Today
+//
+//  Created by Alexey Globchastyy on 15/09/14.
+//  Copyright (c) 2014 Alexey Globchastyy. All rights reserved.
+//
+import UIKit
+import Accelerate
+
+public extension UIImage {
+    public func applyLightEffect() -> UIImage? {
+        return applyBlurWithRadius(30, tintColor: UIColor(white: 1.0, alpha: 0.3), saturationDeltaFactor: 1.8)
+    }
+    
+    public func applyExtraLightEffect() -> UIImage? {
+        return applyBlurWithRadius(20, tintColor: UIColor(white: 0.97, alpha: 0.82), saturationDeltaFactor: 1.8)
+    }
+    
+    public func applyDarkEffect() -> UIImage? {
+        return applyBlurWithRadius(20, tintColor: UIColor(white: 0.11, alpha: 0.73), saturationDeltaFactor: 1.8)
+    }
+    
+    public func applyTintEffectWithColor(_ tintColor: UIColor) -> UIImage? {
+        let effectColorAlpha: CGFloat = 0.6
+        var effectColor = tintColor
+        
+        let componentCount = tintColor.cgColor.numberOfComponents
+        
+        if componentCount == 2 {
+            var b: CGFloat = 0
+            if tintColor.getWhite(&b, alpha: nil) {
+                effectColor = UIColor(white: b, alpha: effectColorAlpha)
+            }
+        } else {
+            var red: CGFloat = 0
+            var green: CGFloat = 0
+            var blue: CGFloat = 0
+            
+            if tintColor.getRed(&red, green: &green, blue: &blue, alpha: nil) {
+                effectColor = UIColor(red: red, green: green, blue: blue, alpha: effectColorAlpha)
+            }
+        }
+        
+        return applyBlurWithRadius(10, tintColor: effectColor, saturationDeltaFactor: -1.0, maskImage: nil)
+    }
+    
+    public func applyBlurWithRadius(_ blurRadius: CGFloat,
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/ec53edf088a4480cb2500e46be0b986e72691277/Headlines/src/Categories/UIImageEffects.swift#L131"></span>Function should have complexity 10 or less: currently complexity equals 17</td>
    </tr>
  </tbody>
</table>